File: php/src/Model/TaskCheckReport.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A dry run: what would happen to this task. Creates and reserves nothing.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TaskCheckReport
{
    public function __construct(
        /** @var list<TaskCheckIssue> */
        public readonly array $issues,
        public readonly int $eligibleWorkerCount,
        /** @var list<string> */
        public readonly array $uncoveredTags,
        /** Epoch-ms */
        public readonly int $evaluatedAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            issues: Json::parseEach($data, 'issues', [TaskCheckIssue::class, 'fromArray']),
            eligibleWorkerCount: (int) ($data['eligibleWorkerCount'] ?? 0),
            uncoveredTags: \array_values(\array_map(strval(...), $data['uncoveredTags'] ?? [])),
            evaluatedAt: (int) ($data['evaluatedAt'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TaskCheckReport.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A dry run: what would happen to this task. Creates and reserves nothing.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TaskCheckReport {
    private List<TaskCheckIssue> issues;
    private Integer eligibleWorkerCount;
    private List<String> uncoveredTags;
    private Long evaluatedAt;

    public List<TaskCheckIssue> getIssues() { return issues; }
    public Integer getEligibleWorkerCount() { return eligibleWorkerCount; }
    public List<String> getUncoveredTags() { return uncoveredTags; }
    public Long getEvaluatedAt() { return evaluatedAt; }
}
